### ISSUE TITLE
The Drop Dead Cannon Game gets absolutely broken!

### ISSUE THREAD

**Issue by DestructMarble250** [ruffle-rs/ruffle#12408]
### Describe the bug

Hello, i was going to play some Drop Dead Cannon, and it looks like it's absolutely broken, when i pressed the play button, it turns into just a black screen with some broken user interface

how to reproduce the steps?
1. go to the following link: [https://www.kongregate.com/games/ttursas/drop-dead-cannon](url)
2. click the play button (after it's been loaded and when you start to play it)
3. play around with the tools and stuff.

here's the full log:

# Player Info
Allows script access: true
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "Radeon HD 3200 Graphics"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(TEXTURE_COMPRESSION_BC | TEXTURE_COMPRESSION_ETC2 | TEXTURE_COMPRESSION_ASTC | TEXTURE_COMPRESSION_ASTC_HDR | CLEAR_TEXTURE)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://game189680.konggames.com/games/ttursas/drop-dead-cannon/frame/4a0c532e-a1af-4503-acb7-1ae47fea8f57/?kongregate_host=www.kongregate.com
SWF URL: https://game189680.konggames.com/gamez/0018/9680/live/DropDeadCannon-Kongregate.swf?kongregate_game_version=1382704309

# Browser Info
User Agent: Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0
Platform: Linux x86_64
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-07-29
Channel: nightly
Built: 2023-07-29T00:19:39.173Z
Commit: 82b88ebff59f7992eeb4d10a18dec28d8714d338
Is extension: false

# Metadata
width: 600
height: 600
frameRate: 60
numFrames: 2
swfVersion: 11
backgroundColor: undefined
isActionScript3: true
uncompressedLength: 6929696

### Expected behavior

I've expected for the game to work correctly

### Affected platform

Browser's extension

### Operating system

Linux Mint 21.1 Cinnamon

### Browser

Mozilla Firefox

### Additional information

If you tried to fix the problem, please reply to me!

**Comment by Lord-McSweeney** [comment]
I couldn't test this because it got too laggy, but earlier in the SWF, I encountered this:
![image](https://github.com/ruffle-rs/ruffle/assets/84632019/1649230f-1541-4ab5-bd2e-67bd5b3fac03)

(the glitchy looking START button)

This seems similar to https://github.com/ruffle-rs/ruffle/issues/12136, but I couldn't check the expected FP output since the game is sitelocked and I couldn't figure out how to break the sitelock.

**Comment by DestructMarble250** [comment]
Yeah, the site-locked thing also happened on the first Cover Orange game and most of TTursas' flash games, but the editor just was a pure dark and black screen with the glichy looking interface (the bottom of it was okay)

![Captura de pantalla de 2023-07-29 16-14-44](https://github.com/ruffle-rs/ruffle/assets/115897739/40ff0887-6eee-434b-9956-559d85bba885)

**Comment by Lord-McSweeney** [comment]
hmm, i got to that screen fine
